FirebaseAdmin/FirebaseAdmin/AppCheck/AppCheckApiClient.cs

```diff
@@ -1,198 +1,288 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.Net.Http;
-using System.Text;
-using System.Text.RegularExpressions;
-using System.Threading;
 using System.Threading.Tasks;
-using FirebaseAdmin.Auth;
-using Newtonsoft.Json;
-using Newtonsoft.Json.Linq;
-using static Google.Apis.Requests.BatchRequest;
+using FirebaseAdmin.Util;
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Http;
+using Google.Apis.Json;
+using Google.Apis.Util;
 
-namespace FirebaseAdmin.Check
+namespace FirebaseAdmin.AppCheck
 {
     /// <summary>
     /// Class that facilitates sending requests to the Firebase App Check backend API.
     /// </summary>
-    /// <returns>A task that completes with the creation of a new App Check token.</returns>
-    /// <exception cref="ArgumentNullException">Thrown if an error occurs while creating the custom token.</exception>
-    /// <value>The Firebase app instance.</value>
-    internal class AppCheckApiClient
+    internal sealed class AppCheckApiClient : IDisposable
     {
-        private const string ApiUrlFormat = "https://firebaseappcheck.googleapis.com/v1/projects/{projectId}/apps/{appId}:exchangeCustomToken";
+        private const string AppCheckUrlFormat = "https://firebaseappcheck.googleapis.com/v1/projects/{projectId}/apps/{appId}:exchangeCustomToken";
         private const string OneTimeUseTokenVerificationUrlFormat = "https://firebaseappcheck.googleapis.com/v1beta/projects/{projectId}:verifyAppCheckToken";
-        private readonly FirebaseApp app;
-        private string projectId;
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="AppCheckApiClient"/> class.
-        /// </summary>
-        /// <param name="value"> Initailize FirebaseApp. </param>
-        public AppCheckApiClient(FirebaseApp value)
+        private readonly ErrorHandlingHttpClient<FirebaseAppCheckException> httpClient;
+        private readonly string projectId;
+
+        internal AppCheckApiClient(Args args)
+        {
+            string noProjectId = "Project ID is required to access app check service. Use a service account "
```

Comment:
Why putting it here as a string and not just in the if statement where it goes wrong?

Reply:
I think that it is good suggestion, I move string in if statement.